### PR TITLE
M3-5873: Build `api-v4` and `validation` with `tsup` with backwards compatibility

### PR DESCRIPTION
**Pull request by bnussman**
## ✨ Description

- Use `tsup` for building `@linode/api-v4` and `@linode/validation`

## 📝 Details
- Deprecates `yarn up` in favor of `yarn dev`
- Runs a dev server for `@linode/validation` so it rebuilds automatically when you are using `yarn dev` / `yarn up`
- Changes any `src/` prefixed imports in `@linode/api-v4` and `@linode/validation` to be relative path imports (`../`)
  - We should do this to keeps those packages simple. Aliases can causes extra complexity and is not beneficial in small packages like ours.
- Configure Cloud Manger's `jest` to use `@linode/api-v4` and `@linode/validation`'s `src`instead of `lib`.
  - We do this because jest was having trouble resolving exports because of its lackluster esm or exports support. 
- Type generation is handled by `tsc`
- Use `incremental` tsc compilation for  `@linode/api-v4` and `@linode/validation` for faster typechecking
- We are going to publish `esm`, `commonjs`, and `iife` as 3 bundled formats
  - This results in 3 respective `index.js`s for each format (excluding type files)
  - Cloud Manager consumes `esm`
  - Jest will consume `commonjs`
  - Dependencies are only bundled into the index.js file for the `iife` format so it is suitable for web use

## 🧪 How to test

- Test all of Cloud Manager, `@linode/api-v4`, and `@linode/validation`
- Run `yarn` from root dir to update dependencies
- run `rm -rf packages/api-v4/lib && rm -rf packages/validation/lib && rm packages/api-v4/index.js && rm packages/api-v4/index.d.ts && rm packages/validation/index.js && rm packages/validation/index.d.ts` to clear old builds of `api-v4` and `validation`
- run `yarn up`

### CODE REVIEW

File: packages/manager/package.json [hunk 1]
```diff
@@ -282,7 +282,9 @@
       "\\.svg$": "<rootDir>/src/components/NullComponent",
       "^react-native$": "react-native-web",
       "ramda": "ramda/src/index.js",
-      "^src/(.*)": "<rootDir>/src/$1"
+      "^src/(.*)": "<rootDir>/src/$1",
+      "@linode/api-v4/lib(.*)$": "<rootDir>/../api-v4/src/$1",
+      "@linode/validation/lib(.*)$": "<rootDir>/../validation/src/$1"
```
Comment (bnussman) [inline packages/manager/package.json:287]: This _should_  be able to be removed after we update Jest. Newer versions of Jest support the `exports` values in the `package.json`, and hopefully Jest will pull the required `commonjs` files

**Review by jdamore-linode — COMMENTED** [review]
I'm gonna finish reviewing this later, but so far I'm liking everything I see! Nice work!

File: packages/api-v4/src/databases/databases.ts [hunk 2]
```diff
@@ -2,14 +2,14 @@ import {
   createDatabaseSchema,
   updateDatabaseSchema,
 } from '@linode/validation/lib/databases.schema';
-import { BETA_API_ROOT as API_ROOT } from 'src/constants';
+import { BETA_API_ROOT as API_ROOT } from '../constants';
```
[the remark this answers is not in the document]
Comment (bnussman) [reply packages/api-v4/src/databases/databases.ts:5]: tsup handled it fine, it actually was a problem with the temporary jest change above

File: packages/api-v4/package.json [hunk 3]
```diff
@@ -16,22 +17,38 @@
...
     "querystring": "^0.2.0",
     "yup": "^0.32.9"
   },
   "scripts": {
-    "start": "concurrently --raw \"tsc -w --preserveWatchOutput\" \"babel src --watch --out-dir lib --extensions '.ts,.tsx'\"",
-    "build": "tsc && babel src --out-dir lib --extensions '.ts,.tsx' && webpack",
-    "build:node": "yarn build --config webpack.node.config.js",
+    "start": "concurrently --raw \"tsc -w --preserveWatchOutput\" \"tsup --watch\"",
+    "build": "concurrently --raw \"tsc\" \"tsup\"",
+    "build:node": "echo not needed",
```
[the remark this answers is not in the document]
Comment (bnussman) [reply packages/api-v4/package.json:51]: I was worried our internal pipeline used it, but it appears like it does not. I'll remove it.

File: packages/api-v4/src/databases/databases.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jdamore-linode) [reply packages/api-v4/src/databases/databases.ts:5]: Ah that makes sense. Thanks!

**Review by jdamore-linode — APPROVED** [review]
Really, really nice! Great work @bnussman

**Review by DevDW — COMMENTED** [review]
Going to finish testing this in-depth first thing tomorrow but for now I've confirmed that the app compiles and Cloud appears to be functioning well

File: packages/manager/package.json [hunk 4]
```diff
@@ -16,8 +16,8 @@
     "url": "https://github.com/Linode/manager.git"
   },
   "dependencies": {
-    "@linode/api-v4": "^0.75.0",
-    "@linode/validation": "0.12.0",
+    "@linode/api-v4": "*",
+    "@linode/validation": "*",
```
Comment (DevDW) [inline packages/manager/package.json:20]: Going to resurface this question https://github.com/linode/manager/pull/8447#discussion_r930133473. Were you able to find any documentation? Is yarn going to pull the correct versions as specified in the corresponding `package.json` files?

**Review by DevDW — APPROVED** [review]
Great job! Resurfaced a question and once this gets released, some updates will need to be made to https://www.npmjs.com/package/@linode/api-v4 and maybe https://www.npmjs.com/package/@linode/validation to keep them accurate.

File: packages/manager/package.json (hunk 4 above)
Comment (bnussman) [reply packages/manager/package.json:20]: I did a bit of looking and didn't find anything super great, but here is the docs on how versioning with the `*` works. 

https://classic.yarnpkg.com/lang/en/docs/dependency-versions/#toc-x-ranges

I found some looking around and people say that yarn will resolve the `*` to the latest version of a dependency and will check a monorepo's workspaces first

https://blog.charlesloubao.com/one-line-javascript-tip-1-how-to-install-a-local/

**Pull request merged**